### DISCUSSION TITLE
Syntax Suggestions

### DISCUSSION THREAD

**Discussion by b4apple** [pikelet-lang/pikelet#250] [Ideas]
Hi team,
I was really impressed by this initiative of making a new programming language and would soon like to see a practical usable pikelet edition. However, going through the files in the example directory, I had a couple of things which I would like to share with you. Using clusters of symbols as a part of the syntax _might_ put off beginners and might defer Pikelet as a first language. Overall, if the syntax bore resemblance to some existing language, chances of it getting more traction become more.
Regards,
Aritro Shome 
Contact me at aritroshome.mpb@gmail.com

**Comment by sullyj3** [comment]
Seems to me like it bears resemblance to several ML derivatives :)

Additionally, beginner friendliness is desirable all else equal, but it seems conceivable and valid for a language to choose to prioritize other desiderata.

**Comment by brendanzab** [comment]
Ooh, yeah curious to know your thoughts. As @sullyj3 says, it's pretty inspired by ML derivatives, but I'm definitely have things I want to change syntax wise and have not set much in stone! But yeah, I think it will probably always have a ML/Type Theory flavour to it.

**Reply by b4apple** [reply]
Can i use ur repo to learn about making programming languages ? or is it not beginner friendly ?

**Reply by brendanzab** [reply]
It might not be the best place to start! 😅 We have a [Discord server](https://discord.gg/yqWzmkV) for the [/r/ProgrammingLanguages](https://www.reddit.com/r/ProgrammingLanguages/) subreddit which might be helpful (it has a #beginners channel). You might also find [Crafting Interpreters](https://craftinginterpreters.com/) helpful, but it might depend on the kind of language you want to learn to make.

**Reply by brendanzab** [reply]
I think there are also some cool courses online from various universities too, which could be fun to look at!

**Reply by b4apple** [reply]
Thanks @brendanzab. Hope to come up with a pikelet counterpart soon !

**Reply by b4apple** [reply]
> I think there are also some cool courses online from various universities too, which could be fun to look at!

Didn't find any :(
